Azure/sonic-buildimage
# sfp: add initial support for CMIS on Quanta IX9

Signed-off-by: Dante Su <dante.su@broadcom.com>



#### Why I did it

Add initial support for CMIS application initialization and loopback
HLD: https://github.com/Azure/SONiC/pull/876

#### How I did it

sonic-buildimage: Update the sfp.py of Quanta IX9 to use the new sfp-refactoring framework

#### How to verify it

1. Bring up Quanta IX9 with the default configuration and CMIS 4.0/5.0 QSFPDD modules
2. Issue 'show logging xcvrd | grep CMIS' to check if the application initialization succeeded
3. Issue 'show interfaces transceiver eeprom --dom' to verify both the static and DOM information parsers
4. Issue 'sudo sfputil show eeprom --dom' to verify both the static and DOM information parsers
5. Issue 'sudo sfputil show error-status -hw' to verify error states

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog

<table>
<tr>
<th>Repo</th>
<th>PR</th>
</tr>
<tr>
<td>SONiC (HLD)</td>
<td>https://github.com/Azure/SONiC/pull/876</td>
</tr>
<tr>
<td>sonic-buildimage</td>
<td>https://github.com/Azure/sonic-buildimage/pull/8899</td>
</tr>
<tr>
<td>sonic-platform-common</td>
<td>https://github.com/Azure/sonic-platform-common/pull/219</td>
</tr>
<tr>
<td>sonic-platform-daemon</td>
<td>https://github.com/Azure/sonic-platform-daemons/pull/217</td>
</tr>
<td>sonic-utilities</td>
<td>https://github.com/Azure/sonic-utilities/pull/1855</td>
</table>

#### A picture of a cute animal (not mandatory but encouraged)
